presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -429,14 +433,14 @@ private SubqueryExpression uncoercedSubquery(SubqueryExpression subquery)
 
     private PlanBuilder appendApplyNode(
             PlanBuilder subPlan,
-            Node subquery,
+            String subqueryError,
             PlanNode subqueryNode,
             Assignments subqueryAssignments,
             boolean correlationAllowed)
     {
         Map<Expression, Expression> correlation = extractCorrelation(subPlan, subqueryNode);
         if (!correlationAllowed && !correlation.isEmpty()) {
-            throw notSupportedException(subquery, "Correlated subquery in given context");
```

Comment:
I think it's ok to keep the original exception given `ApplyNode` has not yet been created.